static/app/components/carousel.tsx

```diff
@@ -0,0 +1,148 @@
+import {useEffect, useRef, useState} from 'react';
+import styled from '@emotion/styled';
+
+import {Button} from 'sentry/components/button';
+import {IconArrow} from 'sentry/icons';
+
+type Props = {
+  children?: React.ReactNode;
+};
+
+function Carousel({children}: Props) {
+  const ref = useRef<HTMLDivElement | null>(null);
+  const [isAtStart, setIsAtStart] = useState<boolean>(true);
+  const [isAtEnd, setIsAtEnd] = useState<boolean>(false);
+  const [showArrows, setShowArrows] = useState<boolean>(true);
+
+  const setStartAndEnd = (scrollLeft: number, totalScrollDist: number) => {
+    // scrollLeft: the amount scrolled horizontally.
+    // start at 0 and end at totalScrollDist
+
+    // totalScrollDist = scrollWidth - clientWidth
+    // scrollWidth: total element width including overflow
+    // clientWidth: width of the element excluding overflow
+    if (scrollLeft <= 0) {
+      setIsAtStart(true);
+      setIsAtEnd(false);
+    } else if (scrollLeft >= totalScrollDist) {
+      setIsAtStart(false);
+      setIsAtEnd(true);
+    } else {
+      setIsAtStart(false);
+      setIsAtEnd(false);
+    }
+  };
+
+  useEffect(() => {
+    if (!ref.current) {
+      return;
+    }
+    const contentWidth = ref.current.scrollWidth - ref.current.clientWidth;
+    if (contentWidth === 0) {
+      setShowArrows(false);
+    }
+
+    const scrollLeft = ref.current.scrollLeft;
+    setStartAndEnd(scrollLeft, contentWidth);
+  }, [ref]);
+
+  const handleScroll = (direction: string) => {
+    requestAnimationFrame(() => {
+      if (!ref.current) {
+        return;
+      }
+
+      const scrollLeft = ref.current.scrollLeft;
+      const itemWidth = parseInt(getComputedStyle(ref.current.children[0]).width, 10);
+
+      let scrollDist = 0;
+      if (direction === 'left') {
+        scrollDist = scrollLeft - itemWidth;
+      } else if (direction === 'right') {
+        scrollDist = scrollLeft + itemWidth;
+      }
+      ref.current.scrollLeft = scrollDist;
```

Comment:
I might use the [`scroll()` API](https://developer.mozilla.org/en-US/docs/Web/API/Element/scroll) with the `smooth` option, so it's clear it has that behavior, instead of using the CSS property